ansible/playbooks/roles/compatibility_layer/defaults/main.yml

```diff
@@ -16,24 +16,17 @@ gentoo_prefix_path: /cvmfs/{{ cvmfs_repository }}/versions/{{ eessi_version }}/c
 gentoo_git_repo: https://github.com/gentoo/gentoo.git
 # Select a specific commit in the gentoo_git_repo that should be used for the bootstrap,
 # e.g. by checking: https://github.com/gentoo/gentoo/commits/master
-# April 17 (29492845e41ea6a0a4a9769c7e0ce287d106079b) commit is after fix for Lmod
-# gentoo_git_commit: 29492845e41ea6a0a4a9769c7e0ce287d106079b
-# June 8 (aab8473aa90e0287553b3348a5c5b17872df4b7b) commit that was current when fetching luaposix
-gentoo_git_commit: aab8473aa90e0287553b3348a5c5b17872df4b7b
+gentoo_git_commit: db21a802e879f713cdb8a80235c5982ca257b63f
 prefix_required_space: 15 GB
 prefix_user_defined_trusted_dirs:
   - "/cvmfs/{{ cvmfs_repository }}/host_injections/{{ eessi_version }}/compat/{{ eessi_host_os }}/{{ eessi_host_arch }}/lib"
 prefix_mask_packages: |
-  # stick to GCC 10.x; using a too recent compiler in the compat layer complicates stuff in the software layer,
+  # stick to GCC 11.x; using a too recent compiler in the compat layer may complicate stuff in the software layer,
```

Comment:
I would go with at least GCC 13, perhaps GCC 14 (unless that causes trouble to build GCC 13.x in software layer)